be/src/formats/orc/orc_chunk_writer.cpp

```diff
@@ -0,0 +1,736 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "formats/orc/orc_chunk_writer.h"
+
+#include "column/array_column.h"
+#include "column/chunk.h"
+#include "column/map_column.h"
+#include "column/struct_column.h"
+#include "gutil/strings/substitute.h"
+
+namespace starrocks {
+
+/*
+** ORCOutputStream
+*/
+OrcOutputStream::OrcOutputStream(std::unique_ptr<starrocks::WritableFile> wfile) : _wfile(std::move(wfile)) {}
+
+OrcOutputStream::~OrcOutputStream() {
+    if (!_is_closed) {
+        close();
+    }
+}
+
+uint64_t OrcOutputStream::getLength() const {
+    return _wfile->size();
+}
+
+uint64_t OrcOutputStream::getNaturalWriteSize() const {
+    return config::vector_chunk_size;
+}
+
+const std::string& OrcOutputStream::getName() const {
+    return _wfile->filename();
+}
+
+void OrcOutputStream::write(const void* buf, size_t length) {
+    if (_is_closed) {
+        throw "The output stream is closed but there are still inputs";
+    }
+    const char* ch = reinterpret_cast<const char*>(buf);
+    Status st = _wfile->append(Slice(ch, length));
+    if (!st.ok()) {
+        throw "write to orc failed: " + st.to_string();
+    }
+    return;
+}
+
+void OrcOutputStream::close() {
+    if (_is_closed) {
+        throw "The output stream is already closed";
+    }
+    Status st = _wfile->close();
+    if (!st.ok()) {
+        throw "close orc output stream failed: " + st.to_string();
+    }
+    _is_closed = true;
+    return;
+}
+
+OrcChunkWriter::OrcChunkWriter(std::unique_ptr<WritableFile> writable_file,
+                               std::shared_ptr<orc::WriterOptions> writer_options, std::shared_ptr<orc::Type> schema,
+                               const std::vector<ExprContext*>& output_expr_ctxs)
+        : _output_stream(std::move(writable_file)) {
+    _type_descs.reserve(output_expr_ctxs.size());
+    for (auto expr : output_expr_ctxs) {
+        _type_descs.push_back(expr->root()->type());
+    }
+    _writer_options = writer_options;
+    _schema = schema;
+}
+
+Status OrcChunkWriter::set_compression(const TCompressionType::type& compression_type) {
+    switch (compression_type) {
+    case TCompressionType::SNAPPY: {
+        _writer_options->setCompression(orc::CompressionKind::CompressionKind_SNAPPY);
+        break;
+    }
+    case TCompressionType::ZLIB: {
+        _writer_options->setCompression(orc::CompressionKind::CompressionKind_ZLIB);
+        break;
+    }
+    case TCompressionType::ZSTD: {
+        _writer_options->setCompression(orc::CompressionKind::CompressionKind_ZSTD);
+        break;
+    }
+    case TCompressionType::LZ4: {
+        _writer_options->setCompression(orc::CompressionKind::CompressionKind_LZ4);
+        break;
+    }
+    case TCompressionType::LZO: {
```

Comment:
does orc support lzo?

Reply:
It seems that orc only supports decompress from lzo and doesn't support compress to lzo. I will fix it in the next pr.